compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -6656,6 +6662,12 @@ private BLangStatementExpression desugarObjectTypeInit(BLangTypeInit typeInitExp
         // init() returning nil is the common case and the type test is not needed for it.
         if (initInvocation.getBType().tag == TypeTags.NIL) {
             initInvocation.name.value = GENERATED_INIT_SUFFIX.value;
+            BLangNode parent = initInvocation.parent;
+            if (parent.getKind() == NodeKind.OBJECT_CTOR_EXPRESSION) {
```

Comment:
```ballerina
listener ABC ep = new;

service on ep {


    resource function get foo(string b) {
    }

    resource function get bar(string b) {
    }
}

public class ABC {

    public function 'start() returns error? {
        error e = error("startError");
        return e;
    }

    public function gracefulStop() returns error? {
        return ();
    }

    public function immediateStop() returns error? {
        return ();
    }

    public function attach(service object {} s, string[]|string? name = ()) returns error? {
        return ();
    }

    public function detach(service object {} s) returns error? {
    }
}

```
above program crashes with
```
[2022-01-05 01:59:06,759] SEVERE {b7a.log.crash} - null 
java.lang.NullPointerException
        at org.wso2.ballerinalang.compiler.desugar.Desugar.desugarObjectTypeInit(Desugar.java:6668)
        at org.wso2.ballerinalang.compiler.desugar.Desugar.visit(Desugar.java:6639)
        at org.wso2.ballerinalang.compiler.tree.expressions.BLangTypeInit.accept(BLangTypeInit.java:65)
        at org.wso2.ballerinalang.compiler.desugar.Desugar.rewriteExpr(Desugar.java:8486)
        at org.wso2.ballerinalang.compiler.desugar.Desugar.visit(Desugar.java:2465)
        at org.wso2.ballerinalang.compiler.tree.statements.BLangAssignment.accept(BLangAssignment.java:81)
        at org.wso2.ballerinalang.compiler.desugar.Desugar.rewrite(Desugar.java:8461)
        at org.wso2.ballerinalang.compiler.desugar.Desugar.rewrite(Desugar.java:8499)
        at org.wso2.ballerinalang.compiler.desugar.Desugar.rewriteStmt(Desugar.java:8505)
        at org.wso2.ballerinalang.compiler.desugar.Desugar.visit(Desugar.java:1328)
        at org.wso2.ballerinalang.compiler.tree.BLangBlockFunctionBody.accept(BLangBlockFunctionBody.java:60)
        at org.wso2.ballerinalang.compiler.desugar.Desugar.rewrite(Desugar.java:8461)
        at org.wso2.ballerinalang.compiler.desugar.Desugar.visit(Desugar.java:1370)
        at org.wso2.ballerinalang.compiler.tree.BLangFunction.accept(BLangFunction.java:76)
        at org.wso2.ballerinalang.compiler.desugar.Desugar.rewrite(Desugar.java:8461)
        at org.wso2.ballerinalang.compiler.desugar.Desugar.splitInitFunction(Desugar.java:10367)
        at org.wso2.ballerinalang.compiler.desugar.Desugar.visit(Desugar.java:820)
        at org.wso2.ballerinalang.compiler.tree.BLangPackage.accept(BLangPackage.java:167)
        at org.wso2.ballerinalang.compiler.desugar.Desugar.rewrite(Desugar.java:8461)
        at org.wso2.ballerinalang.compiler.desugar.Desugar.perform(Desugar.java:458)
        at io.ballerina.projects.internal.CompilerPhaseRunner.desugar(CompilerPhaseRunner.java:216)
        at io.ballerina.projects.internal.CompilerPhaseRunner.performBirGenPhases(CompilerPhaseRunner.java:152)
        at io.ballerina.projects.ModuleContext.generateCodeInternal(ModuleContext.java:453)
        at io.ballerina.projects.ModuleCompilationState$4.generatePlatformSpecificCode(ModuleCompilationState.java:132)
        at io.ballerina.projects.ModuleContext.generatePlatformSpecificCode(ModuleContext.java:390)
        at io.ballerina.projects.JBallerinaBackend.performCodeGen(JBallerinaBackend.java:159)
        at io.ballerina.projects.JBallerinaBackend.<init>(JBallerinaBackend.java:137)
        at io.ballerina.projects.JBallerinaBackend.lambda$from$0(JBallerinaBackend.java:113)
        at java.base/java.util.HashMap.computeIfAbsent(HashMap.java:1134)
        at io.ballerina.projects.PackageCompilation.getCompilerBackend(PackageCompilation.java:182)

```

Reply:
This can be fix by adding null check(`if (parent != null && parent.getKind() == NodeKind.OBJECT_CTOR_EXPRESSION`)
Is it okay? Can the parent of `initInvocation` be null?

Reply:
Yes